src/vs/vscode.d.ts

```diff
@@ -1534,12 +1534,31 @@ declare module 'vscode' {
 		 */
 		ignoreFocusOut?: boolean;
 
+		/**
+		 * An optional flag to make the picker multi-select, if true the result is an array of picks.
+		 */
+		multiSelect?: boolean;
+
 		/**
 		 * An optional function that is invoked whenever an item is selected.
 		 */
 		onDidSelectItem?(item: QuickPickItem | string): any;
 	}
 
+	/**
+	 * Represents an item that can be selected in a multi-select quick pick UI.
+	 */
+	export interface MultiSelectQuickPickItem extends QuickPickItem {
+		selected?: boolean;
+	}
```

Comment:
I wouldn't use a separate interface but have it on `QuickPickItem` and document (btw jsdoc is missing) that `selected` is only honoured in the must-select case.